src/WebJobs.Extensions.DurableTask/Listener/TaskActivityShim.cs

```diff
@@ -54,7 +54,28 @@ public override async Task<string> RunAsync(TaskContext context, string rawInput
                 functionType: FunctionType.Activity,
                 isReplay: false);
 
-            FunctionResult result = await this.executor.TryExecuteAsync(triggerInput, CancellationToken.None);
+            FunctionResult result;
+            try
+            {
+                result = await this.executor.TryExecuteAsync(triggerInput, CancellationToken.None);
```

Comment:
Previously we did not catch or trace exceptions thrown by `TryExecuteAsync`, which is why the ObjectDisposedException problem was so hard to diagnose. When a function fails with an exception, this still returns by with a `FunctionResult.Succeeded` value set to `false`. Now we look for both cases. An exception means that something unexpected failed in WebJobs.

Reply:
Agreed on both points.